### PR TITLE
Do not use FileOrCreate resource type.

### PR DESCRIPTION
**Pull request by ereslibre**
The init container for the secrets will create this file.

### CODE REVIEW

File: manifests/public.yaml [hunk 1]
```diff
@@ -89,6 +89,7 @@ spec:
   - name: salt-master-config
     image: sles12/velum:__TAG__
     command: ["sh", "-c", "umask 377;
+                           rmdir /salt-master-credentials/55-returner-credentials.conf;
```
Comment (mjura) [inline manifests/public.yaml]: Why this is directory and not file?

Comment (ereslibre) [reply manifests/public.yaml]: The very first time k8s loads this manifest, this file does not exist on the host path, and is created as a directory. The init container removes this directory and writes the file. It's far from ideal, but does the trick.

**Comment by ereslibre** [comment]
This isn't passing on CI because the orchestration never finishes (but Velum is started with this patch). Now the problem is related to Dex and the update of kubernetes from 1.7.7 to 1.8.5.

File: manifests/public.yaml (hunk 1 above)
Comment (inercia) [reply manifests/public.yaml]: wtf! I think you should add a comment explaining this...

**Pull request merged**